src/main/java/org/terasology/thirst/ThirstAuthoritySystem.java

```diff
@@ -64,6 +117,7 @@
     public void onPlayerSpawn(OnPlayerSpawnedEvent event, EntityRef player,
                               ThirstComponent thirst) {
         resetThirst(player, thirst);
+        delayManager.addPeriodicAction(player, THIRST_DAMAGE_ACTION_ID, 0, healthDecreaseInterval);
```

Comment:
I don't know on which entity should i add the periodic event. Currently, i have added it to player entity.

Reply:
Yup, so adding it to the player is fine. The only purpose of that entity is to allow the system to work. You can even just make a blank entity, set it to never despawn and use that if you wanted.

I think there _might_ be some issues with using player entity in multiplayer but that's something best found by testing.